src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -205,249 +170,45 @@ where
-    >,
-    /// The latest status update for this worker, if any.
-    status_update: Option<HealthStatusUpdate>,
-    /// A list of partitions that this source reader instance
-    /// is sure it doesn't care about. Required so the
-    /// remap operator can eventually determine whether
-    /// a timestamp is closed.
-    unconsumed_partitions: Vec<PartitionId>,
-    /// See `SourceMessageBatch`.
-    source_upper: OffsetAntichain,
-    /// See `SourceMessageBatch`.
-    batch_upper: OffsetAntichain,
-}
+            let health_token = health_operator(scope, config, health_stream);
 
-fn build_source_reader_stream<S>(
-    source_reader: S,
-    config: RawSourceCreationConfig,
-    initial_source_upper: OffsetAntichain,
-    mut source_upper: OffsetAntichain,
-) -> Pin<
-    // TODO(guswynn): determine if this boxing is necessary
-    Box<
-        impl tokio_stream::Stream<Item = Option<SourceReaderOperatorOutput<S::Key, S::Value, S::Diff>>>,
-    >,
->
-where
-    S: SourceReader + 'static,
-{
-    let RawSourceCreationConfig {
-        name: _,
-        id,
-        num_outputs: _,
-        worker_id,
-        worker_count,
-        timestamp_interval,
-        encoding: _,
-        storage_metadata: _,
-        resume_upper: _,
-        base_metrics: _,
-        now: _,
-        persist_clients: _,
-        source_statistics,
-    } = config;
-    Box::pin(async_stream::stream!({
-        // Most recent batch upper frontier, does not regress.
-        // TODO(aljoscha): We track this as we go, but we could also derive it
-        // by iterating over all messages in a batch when we emit it.
-        let mut batch_upper = OffsetAntichain::new();
-
-        // Send along an empty batch, so that the reclock operator knows
-        // about the current frontier. Otherwise, if there are no new
-        // messages after a restart, the reclock operator would be stuck and
-        // not advance its downstream frontier.
-        yield Some(SourceReaderOperatorOutput {
-            messages: HashMap::new(),
-            status_update: None,
-            unconsumed_partitions: Vec::new(),
-            source_upper: initial_source_upper,
-            batch_upper: batch_upper.clone(),
-        });
-
-        let source_stream = source_reader.into_stream(timestamp_interval).fuse();
+            (token, health_token)
+        })
+    };
 
-        tokio::pin!(source_stream);
+    let (remap_stream, remap_token) =
+        remap_operator(scope, config.clone(), source_upper_rx, &resume_stream);
 
-        // Emit batches more frequently than we mint new timestamps. We're
-        // hoping that most of the batches that we emit will end up making
-        // it into the freshly minted bindings when remap_operator ticks.
-        let mut emission_interval = tokio::time::interval(timestamp_interval / 5);
-        emission_interval.set_missed_tick_behavior(MissedTickBehavior::Skip);
+    let ((reclocked_stream, reclocked_err_stream), _reclock_token) =
```

Comment:
It was already unused before. Mostly, because a) it's not a "source", so doesn't have to be teared down, and it doesn't do any dangerous modifications of outside state, which would also have to be guarded by a token.